lib/grp.gi

```diff
@@ -339,6 +339,10 @@ InstallImmediateMethod( IsPerfectGroup,
     0,
     grp -> Size( grp ) = 1 );
 
+InstallMethod( IsPerfectGroup, "for groups having abelian invariants",
+    [ IsGroup and HasAbelianInvariants ],
+    grp -> Length( AbelianInvariants( grp ) ) = 0 );
+
```

Comment:
@fingolfin I did some poking around in the library. As far as I can tell, currently ````AbelianInvariants```` can only be computed for (subgroups of) fp-groups or for all groups, but then an ````IsFinite```` check is immediately forced. (Or it can be computed for Pcp-groups, but that uses the polycyclic package.) So currently I do not see a way to define a group for which we can compute ````AbelianInvariants```` and this line would run before any other for ````IsPerfectGroup````.